### PR TITLE
Harrison/passport/google strat

### PR DESCRIPTION
**Pull request by HarrisonMS**
Added Google strategy passport authentication. and cleaned up merge conflicts

### CODE REVIEW

File: api/routers/auth-router.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// GET route for when you click on login - passport authenticates through google
+router.get(
+	"/google",
+	googleAuth.passport.authenticate("google", {
+		scope: ["openid email profile"],
+	})
+);
+
+//* If successful auth - redirects to home page, if not - redirects to /login
+router.get(
```
Comment (ashley-bergsma) [inline api/routers/auth-router.js:23]: If I'm reading this right, all the Google login routes are going through the "/auth/" router endpoints? And the local Passport routes are server level. Can we combine and abstract these logins together?

File: api/routers/users-model.js [hunk 2]
```diff
@@ -5,35 +5,36 @@
...
 
 function findById(id) {
-    return db('users')
-        .where({ id })
-        .first();
+	return db("users").where({ id }).first();
+}
+
+function findByGoogleId(id) {
+	return db("users").where({ googleId: id }).first();
```
Comment (ashley-bergsma) [inline api/routers/users-model.js:35]: lol you didn't like multi-line model functions?

File: api/routers/auth-router.js (hunk 1 above)
Comment (HarrisonMS) [reply api/routers/auth-router.js:23]: Yea totally I was just trying to merge them together with the least amount of changes for now. But yes I think that would be ideal and could even add the strategies in the same file so we don't have two separate passport configs

**Review by ashley-bergsma — APPROVED** [review]
Looks good! 

I think the next step will be talking about how these login endpoints should be organized. They're a little all-over the damn place right now.

File: api/routers/users-model.js (hunk 2 above)
Comment (HarrisonMS) [reply api/routers/users-model.js:35]: you mean breaking them on the .where methods and what not.. I didn't even realize I was doing it thought the linter would help me out there haha

**Pull request merged**